pkg/linode-bs/luks.go

```diff
@@ -113,44 +104,48 @@ func getLuksContext(secrets map[string]string, context map[string]string, lifecy
 	}
 }
 
-func luksFormat(source string, ctx LuksContext) error {
+func luksFormat(ctx LuksContext, source string) error {
 	cryptsetupCmd, err := getCryptsetupCmd()
 	if err != nil {
 		return err
 	}
-	filename, err := writeLuksKey(ctx.EncryptionKey)
-	if err != nil {
-		return err
-	}
-
-	defer func() {
-		e := os.Remove(filename)
-		if e != nil {
-			klog.Errorf("cannot delete temporary file %s: %s", filename, e.Error())
-		}
-	}()
 
 	// initialize the luks partition
 	cryptsetupArgs := []string{
 		"-v",
 		"--batch-mode",
 		"--cipher", ctx.EncryptionCipher,
 		"--key-size", ctx.EncryptionKeySize,
-		"--key-file", filename,
+		"--key-file", "-",
 		"luksFormat", source,
 	}
 
 	klog.V(4).Info("executing cryptsetup luksFormat command ", cryptsetupArgs)
 
-	out, err := exec.Command(cryptsetupCmd, cryptsetupArgs...).CombinedOutput()
+	cmd := exec.Command(cryptsetupCmd, cryptsetupArgs...)
+
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stdin pipe for cryptsetup, got err: %s", err)
```

Comment:
Might want either `%v` or `%w` instead of `%s`?

Reply:
imo there's not much value in `%w` as we're not checking the error types further up the stack, and there's not much we can do to handle any of the errors that might come back from `StdinPipe()`. As for `%v` vs `%s` I don't really see the difference here to be honest, as far as I understand calling `%s` on an error just calls `error.Error()` which returns a string.

I would like to understand why would you prefer `%v` though

Reply:
Agree with @AshleyDumaine with `%w`, I feel like this is a good practice to follow.

Reply:
I've made the change to use `%w` - I am interested in the argument for doing so here though as I don't really see it, but I'm open to having my mind changed.